File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/di/module/ViewModelModule.java
package com.kalashnyk.denys.samplemvvmapp.di.module;

import android.app.Application;

import com.kalashnyk.denys.App;
import com.kalashnyk.denys.samplemvvmapp.di.scope.ViewModelScope;
import com.kalashnyk.denys.samplemvvmapp.domain.AllUsersViewModel;
import com.kalashnyk.denys.samplemvvmapp.domain.SingleUserViewModel;
import com.kalashnyk.denys.samplemvvmapp.repository.AppRepository;

import dagger.Module;
import dagger.Provides;

@Module
public class ViewModelModule {

    Application mApp;

    public ViewModelModule(App app) {
        mApp = app;
    }

    @ViewModelScope
    @Provides
    AllUsersViewModel providesAllUserViewModel(AppRepository repository) {
        return new AllUsersViewModel(mApp, repository);
    }

    @ViewModelScope
    @Provides
    SingleUserViewModel providesSingleUserViewModel(AppRepository repository) {
        return new SingleUserViewModel(mApp, repository);
    }
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/di/module/ViewModelModule.kt
package com.kalashnyk.denys.kotlinsample.di.module

import android.app.Application
import com.kalashnyk.denys.kotlinsample.App
import com.kalashnyk.denys.kotlinsample.di.scope.ViewModelScope
import com.kalashnyk.denys.kotlinsample.domain.AllUsersViewModel
import com.kalashnyk.denys.kotlinsample.domain.SingleUserViewModel
import com.kalashnyk.denys.kotlinsample.repository.AppRepository
import dagger.Module
import dagger.Provides

@Module
class ViewModelModule(app: App) {

    internal var app: Application = app

    @ViewModelScope
    @Provides
    internal fun providesAllUserViewModel(repository: AppRepository): AllUsersViewModel {
        return AllUsersViewModel(app, repository)
    }

    @ViewModelScope
    @Provides
    internal fun providesSingleUserViewModel(repository: AppRepository): SingleUserViewModel {
        return SingleUserViewModel(app, repository)
    }
}